File: ucare_backend/src/main/java/com/douzone/ucare/service/PrescriptionService.java
package com.douzone.ucare.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.douzone.ucare.repository.PrescriptionRepository;
import com.douzone.ucare.vo.PrescriptionVo;
import com.douzone.ucare.vo.ReceiptVo;

@Service
public class PrescriptionService {
	
	@Autowired
	private PrescriptionRepository prescriptionRepository;
	
	public int create(PrescriptionVo data) {
		return prescriptionRepository.create(data);
	}
	
	public List<ReceiptVo> retrieveCureYN(String date) {
		return prescriptionRepository.retrieveCureYN(date);
	}
	
	public List<ReceiptVo> retrieveByDiagnosisNo(Long diagnosisNo) {
		return prescriptionRepository.retrieveByDiagnosisNo(diagnosisNo);
	}
	
	public int update(PrescriptionVo data) {
		return prescriptionRepository.update(data);
	}
	
	public int deleteByPrescriptionNo(Long diseaseNo) {
		return prescriptionRepository.deleteByPrescriptionNo(diseaseNo);
	}
}

File: ucare_frontend/src/service/prescriptionService.js
import axios from 'axios';
import { PRESCRIPTION_API_BASE_URL } from './urlConfig'

class prescriptionService {
  create(data){
    return axios.post(PRESCRIPTION_API_BASE_URL + '/create', data, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  retrieveCureYN(date){
    return axios.get(PRESCRIPTION_API_BASE_URL + '/retrieveCureYN/' + date, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  retrieveByDiagnosisNo(diagnosisNo){
    return axios.get(PRESCRIPTION_API_BASE_URL + '/retrieveByDiagnosisNo/' + diagnosisNo, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  update(data){
    return axios.put(PRESCRIPTION_API_BASE_URL + '/update', data, { headers: { Authorization: localStorage.getItem("authorization") }})
  }

  deleteByPrescriptionNo(prescriptionNo){
    return axios.delete(PRESCRIPTION_API_BASE_URL + '/deleteByPrescriptionNo/' + prescriptionNo, { headers: { Authorization: localStorage.getItem("authorization") }});
  }
}

export default new prescriptionService();